lib/vtls/vtls.c

```diff
@@ -570,14 +571,23 @@ CURLcode Curl_ssl_shutdown(struct connectdata *conn, int sockindex)
  */
 CURLcode Curl_ssl_set_engine(struct Curl_easy *data, const char *engine)
 {
-  return Curl_ssl->set_engine(data, engine);
+  CURLcode result = Curl_ssl->set_engine(data, engine);
+  if(result)
+    return result;
+
+  /* store engine name, to be reinstanced on copied handles */
+  return Curl_setstropt(&data->set.str[STRING_SSL_ENGINE], engine);
```

Comment:
I think I would prefer if `Curl_setstropt` is used in setopt.c to store the string, as then Curl_ssl_set_engine() can remain untouched and can be used fine from within `curl_easy_duphandle` too.